File: src/main/java/frc/team4069/robot/subsystems/VacuumSubsystem.java
package frc.team4069.robot.subsystems;

import com.ctre.phoenix.motorcontrol.ControlMode;
import frc.team4069.robot.io.IOMapping;
import frc.team4069.robot.motors.TalonSRXMotor;

public class VacuumSubsystem extends SubsystemBase {

    private static VacuumSubsystem instance;

    private TalonSRXMotor talon;

    private VacuumSubsystem() {
        talon = new TalonSRXMotor(IOMapping.VACUUM_CAN_BUS, true);
    }

    public void start() {
        talon.set(ControlMode.PercentOutput, 1);
    }

    public void stop() {
        talon.stop();
    }

    public boolean isStarted() {
        return talon.isStarted();
    }

    public static VacuumSubsystem getInstance() {
        if (instance == null) {
            instance = new VacuumSubsystem();
        }

        return instance;
    }
}

File: src/main/java/frc/team4069/robot/subsystems/VacuumSubsystem.kt
package frc.team4069.robot.subsystems

import frc.team4069.robot.io.IOMapping
import frc.team4069.robot.motors.TalonMotor

object VacuumSubsystem : SubsystemBase() {

    private val talon = TalonMotor(IOMapping.VACUUM_CAN_BUS, reversed = true)

    fun reset() {
        talon.stop()
    }

    /**
     * Starts the vacuum motor
     */
    fun start() = talon.setConstantSpeed(1.0)

    /**
     * Stops the vacuum motor
     */
    fun stop() = talon.stop()

    fun isStarted(): Boolean = talon.isStarted()
}